compiler/builtins/string.ts

```diff
@@ -1275,7 +1275,43 @@ i32.store8 0 12`;
       Porffor.wasm.i32.store16(Porffor.wasm`local.get ${tmp}` + tmpLen * 2, x, 0, 4);
       tmpLen++;
     }
-  } else {
+  } 
+  else if (sepLen === 0) {
+    const clammedLimit: i32 = limit > thisLen ? thisLen : limit;
+    for (let i = 0; i <= clammedLimit; i+=2) {
+      tmp = Porffor.allocate();
+      tmpLen = 0;
+      const x: i32 = Porffor.wasm.i32.load16_u(Porffor.wasm`local.get ${_this}` + i, 0, 4);
+
+      Porffor.wasm.i32.store16(Porffor.wasm`local.get ${tmp}` + tmpLen, x, 0, 4);
+      tmpLen = 1;
+      tmp.length = tmpLen;
+      out.length = outLen;
+
+Porffor.wasm`
+  local.get ${out}
+  local.get ${outLen}
+  i32.const 9
+  i32.mul 
+  i32.add
+  local.get ${tmp}
+  f64.convert_i32_u
+  f64.store 0 4
+
+  local.get ${out}
+  local.get ${outLen}
+  i32.const 9 
+  i32.mul
+  i32.add
+  local.get ${sType}
+  i32.store8 0 12
+    `;
+      tmpLen++;
+      outLen++;
+    }
+    return out;
```

Comment:
As we just return out here you can skip setting `tmpLen` etc and just use 0/1 as constants, but thanks for the effort still

Reply:
Thanks. I have removed that, and replaced allocate with allocateBytes